samza-core/src/main/java/org/apache/samza/config/TaskConfig.java

```diff
@@ -147,6 +147,9 @@ public class TaskConfig extends MapConfig {
       "task.transactional.state.retain.existing.state";
   private static final boolean DEFAULT_TRANSACTIONAL_STATE_RETAIN_EXISTING_STATE = true;
 
+  public static final String WATERMARK_IDLE_MS = "task.watermark.idle.ms";
```

Comment:
Can we also add some docs for config too?

Reply:
Thanks for the catch. Add the javadocs above. I didn't add the config to the Samza configuration webpage since the whole watermark in Samza is not fully supported today (no watermark injected from sources). It's a bit confusing to the users to suddenly have this config without being usable. We can consider adding it if we add further support in the sources.